### PR TITLE
Feature/implementation of additional game conditions

### PR DESCRIPTION
**Pull request by Mankur11**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -67,13 +67,25 @@
         return Math.floor(Math.random() * 6) + 1;
     }
 
+    // if the player rolls a 1, all his round score gets lost. after that, it's the next player's turn
+    function isOnePointRolledOnDice() {
+        var state = getGameState();
```
Comment (ostapgv) [inline app.js]: You can combine it to one line `return getGameState().currentDiceScore === 1;`

File: app.js [hunk 2]
```diff
@@ -67,13 +67,25 @@
...
     function rollDice() {
         var currentDiceScore = getRandomValueOfDice();
         setGameState({
             currentDiceScore: currentDiceScore,
             roundDiceScore: getGameState().roundDiceScore + currentDiceScore,
             isActiveGame: true
-        })
+        });
+        if (isOnePointRolledOnDice()){
+            nextPlayer();
```
Comment (ostapgv) [inline app.js]: I'd suggest to use `setTimeout` to show the dice score result at least some time. You could also show some message like `${playerName} rolled 1. The turn goes to the next player.`

File: app.js [hunk 3]
```diff
@@ -67,13 +68,34 @@
         return Math.floor(Math.random() * 6) + 1;
     }
 
+    // if the player rolls a 1, all his round score gets lost. after that, it's the next player's turn
```
Comment (ostapgv) [inline app.js]: The description for the function is not really related to it's functionality (it's just the function that checks if the player have rolled 1 on dice). Such kind of descriptions should be put either to the README or to the place with appropriate logic.

File: app.js [hunk 4]
```diff
@@ -67,13 +68,34 @@
...
             currentDiceScore: currentDiceScore,
             roundDiceScore: getGameState().roundDiceScore + currentDiceScore,
             isActiveGame: true
-        })
+        });
+
+        if (isOnePointRolledOnDice()) {
+            setGameState({
+                isOnePointRolledOnDice: true,
+                isActiveGame: false
```
Comment (ostapgv) [inline app.js]: Does the game end when someone's rolling 1?

File: app.js [hunk 5]
```diff
@@ -67,13 +68,34 @@
...
+            });
+            console.log(getGameState());
```
Comment (ostapgv) [inline app.js]: Cleanup logging please.

File: app.js [hunk 6]
```diff
@@ -67,13 +68,34 @@
...
+            setTimeout(nextPlayer, 2000);
+        } else {
+            setGameState({
+                isOnePointRolledOnDice: false,
+                isActiveGame: true
```
Comment (ostapgv) [inline app.js]: You don't need to set `isActiveGame` here.

File: app.js [hunk 7]
```diff
@@ -67,13 +68,34 @@
...
+            });
+            console.log(getGameState());
```
Comment (ostapgv) [inline app.js]: console.log cleanup as well please.

File: app.js (hunk 3 above)
Comment (Mankur11) [reply app.js]: created README, cut this description from code

File: app.js (hunk 4 above)
Comment (Mankur11) [reply app.js]: no, fixed

File: app.js (hunk 5 above)
Comment (Mankur11) [reply app.js]: have some trouble here, still working

File: app.js (hunk 6 above)
Comment (Mankur11) [reply app.js]: fixed

**Pull request merged**